crates/matrix-sdk-ui/tests/integration/timeline/edit.rs

```diff
@@ -444,6 +445,104 @@ async fn test_send_reply_edit() {
     server.verify().await;
 }
 
+#[async_test]
+async fn test_edit_to_replied_updates_reply() {
+    let room_id = room_id!("!a98sd12bjh:example.org");
+    let (client, server) = logged_in_client_with_server().await;
+    let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
+
+    let mut sync_builder = SyncResponseBuilder::new();
+    sync_builder.add_joined_room(JoinedRoomBuilder::new(room_id));
+
+    mock_sync(&server, sync_builder.build_json_sync_response(), None).await;
+    let _response = client.sync_once(sync_settings.clone()).await.unwrap();
+    server.reset().await;
+
+    mock_encryption_state(&server, false).await;
+
+    let room = client.get_room(room_id).unwrap();
+    let timeline = room.timeline().await.unwrap();
+    let (_, mut timeline_stream) =
+        timeline.subscribe_filter_map(|item| item.as_event().cloned()).await;
+
+    let f = EventFactory::new();
+    let event_id = event_id!("$original_event");
+    let user_id = client.user_id().unwrap();
+
+    // When a room has two messages, one is a reply to the other…
+    sync_builder.add_joined_room(
+        JoinedRoomBuilder::new(room_id)
+            .add_timeline_event(f.text_msg("bonjour").sender(user_id).event_id(event_id))
+            .add_timeline_event(f.text_msg("hi back").reply_to(event_id).sender(*ALICE)),
+    );
```

Comment:
Can you test with 2 events replying to `event_id` please? The idea is to ensure we are looping over all events that are a reply to the edited event.

Reply:
Sure!